### PR TITLE
[EASI-3325] - IT implementation category page and restructure

### PR DESCRIPTION
**Pull request by patrickseguraoddball**
# EASI-3325

## Changes and Description

- Relocated HKC components to all be groups together
- Renamed `HelpAndKnowledgeCategoryTag` to `HelpCategoryTag` to better match other name statndards
- Consolidated `GettingStarted` into a central `AllArticles` component to render all categories
- Prepped `IT implementation` category for upcoming help articles
- Altered the `ArticleTypeProps` type to be more dynamic in conjunction with URL params - `getting-started` | `it-implementation`
- Added/updated unit tests and snapshots

## How to test this change

- Verify HKC works exactly the same as before

## PR Author Review Checklist

- [ ] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [ ] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

File: src/views/HelpAndKnowledge/Articles/_components/HelpCardGroup/index.tsx [hunk 1]
```diff
@@ -2,13 +2,13 @@ import React, { useState } from 'react';
...
 
-import ArticleCard from 'components/ArticleCard';
+import ArticleCard from 'views/HelpAndKnowledge/Articles/_components/ArticleCard';
 
 import helpAndKnowledgeArticles from '../..';
 
 type HelpCardGroupType = {
   className?: string;
-  filter?: 'itImplementation' | 'gettingStarted';
+  filter?: 'it-implementation' | 'getting-started' | string | null;
```
Comment (garyjzhao) [inline src/views/HelpAndKnowledge/Articles/_components/HelpCardGroup/index.tsx:11]: Doesn't adding the `string` type kinda defeats the purpose of explicit spelling out the types `'it-implementation` and `getting-started`?

Comment (patrickseguraoddball) [reply src/views/HelpAndKnowledge/Articles/_components/HelpCardGroup/index.tsx:11]: That param is coming direectly from a URL optional param, so it has to be `string` | `null` to comply.  Not really sure atm how to enforce this

**Review by garyjzhao — CHANGES_REQUESTED** [review]
All looks good. 
Just the nit regarding the HelpCardGroupType filter type, as what we discussed in slack

**Review by garyjzhao — APPROVED** [review]
LGTM

**Pull request merged**